File: math/fibonacci/php/fibonacci.php
<?php
// fib function computes n numbers
// of the fibonacci sequence
function fib(int $n, int $a, int $b) {
    $c = 0;
    for($i = 0; $i < $n; $i++) {
        $c = $a + $b;
        $a = $b;
        $b = $c;
        echo " ";
        echo $c;
    }
}
// exemple for n = 10
fib(10,0,1);
?>

File: math/fibonacci/Rust/fibonacci.rs
use std::cmp;

/// Returns a vector containing the fibonacci sequence, up to n numbers.
fn fib(n: u32) -> Vec<u32> {
    let mut n2 = n;
    let mut f = vec![];

    if n == 0 {
        return f;
    }

    while n2 > cmp::max(n-2, 0) {
        f.push(1);
        n2 -= 1;
    }

    while n2 > 0 {
        let length = f.len();
        let a = f[length-1];
        let b = f[length-2];
        f.push(a+b);
        n2 -= 1;
    }

    f
}
